src/components/cesium_map/cesium_UI.js

```diff
@@ -285,6 +287,8 @@ class CesiumMap extends React.Component {
         moorea.heading,
         moorea.pitch));
     }
+    // force an update of primitives whenever changing view 
+    this.updatePrimitive(cameraLat, cameraLong);
   }
```

Comment:
Even when latitude/longitude does not change (e.g. when clicking Global button), we need to force an update of primitive points to check if the number of points exceed the maximum we have set.